src/libraries/sendtohelix-scenarios.proj

```diff
@@ -0,0 +1,324 @@
+-->
+
+<Project InitialTargets="BuildHelixWorkItems" DefaultTargets="RunInParallelForEachScenario">
+
+  <!-- This project uses the helix SDK, documented at
+       https://github.com/dotnet/arcade/tree/master/src/Microsoft.DotNet.Helix/Sdk,
+       to send test jobs to helix.
+  -->
+
+  <Import Sdk="Microsoft.DotNet.Helix.Sdk" Project="Sdk.props" Condition=" '$(UsesHelixSdk)' == 'true' " />
+
+  <!-- We only import the Helix Sdk if " '$(UsesHelixSdk)' == 'true' ", but Helix auto-magically imports local *.props
+       files. If we're not going to import the Helix Sdk, we still need to import those files so we can reference
+       the properties defined there.
+  -->
+  <Import Project="$(MSBuildThisFileDirectory)/*.props" Condition=" '$(UsesHelixSdk)' != 'true' " />
+
+  <!-- RunInParallelForEachScenario: the "root" target of this file (see description at the top of this file) -->
+  <Target Name="RunInParallelForEachScenario"
+          Condition=" '$(UsesHelixSdk)' != 'true' ">
+
+    <PropertyGroup>
+      <!-- This specifies what properties are needed to be passed down as global properties to a child project invocation. -->
+
+      <_PropertiesToPass>
+        RuntimeFlavor=$(RuntimeFlavor);
+        TargetArchitecture=$(TargetArchitecture);
+        Configuration=$(Configuration);
+        TargetOS=$(TargetOS);
+        TestScope=$(TestScope);
+        TestRunNamePrefixSuffix=$(TestRunNamePrefixSuffix);
+        Creator=$(Creator);
+        HelixAccessToken=$(HelixAccessToken);
+        HelixBuild=$(HelixBuild);
+        HelixTargetQueues=$(HelixTargetQueues);
+        HelixType=$(HelixType);
+        BuildTargetFramework=$(BuildTargetFramework);
+        BuildSettings=$(BuildSettings)
+      </_PropertiesToPass>
+    </PropertyGroup>
+
+    <Error Condition="'$(_Scenarios)' == ''" Text="_Scenarios not set" />
+
+    <Message Importance="High" Text="Using _Scenarios: $(_Scenarios)" />
+    <Message Importance="High" Text="Using Queues: $(HelixTargetQueues)" />
+    <Message Importance="High" Text="BuildSettings: $(BuildSettings)" />
+    <Message Importance="High" Text="BuildTargetFramework: $(BuildTargetFramework)" />
+    <Message Importance="High" Text="TestArchiveTestsRoot: $(TestArchiveTestsRoot)" />
+    <Message Importance="High" Text="TestHostRootPath: $(TestHostRootPath)" />
+    <Message Importance="High" Text="TestArchiveRoot: $(TestArchiveRoot)" />
+    <Message Importance="High" Text="TestArchiveRuntimeRoot: $(TestArchiveRuntimeRoot)" />
+    <Message Importance="High" Text="TestArchiveRuntimeFile: $(TestArchiveRuntimeFile)" />
+
+    <MSBuild Projects="$(MSBuildProjectFile)" Targets="PrepareCorrelationPayloadDirectory" Properties="Scenarios=$(_Scenarios)" />
+
+    <ItemGroup>
+      <_Scenarios Include="$(_Scenarios.Split(','))" />
+
+      <!-- MSBuild creates a new instance of the project for each %(_Scenarios.Identity) and can build them in parallel. -->
+      <_ProjectsToBuild Include="$(MSBuildProjectFile)">
```

Comment:
Why not instead use `sendtohelix.proj`? that way we don't have to duplicate all the helix targets and properties?

Reply:
I would want to avoid so much duplication of the code as, as we move and add more features we would need to maintain both this projects (for example to add more helix data to kusto info, or other stuff).

Reply:
The issue is that my new `sendtohelix-scenarios.proj` is quite a bit more complicated than the existing `sendtohelix.proj`. So I did it as-is to avoid perturbing (and potentially breaking) the existing queues, at least while I was doing development,

I can certainly make everyone use my new system, and have the existing pipelines use a default or empty scenario list. This would require more testing (maybe drastically more testing) to ensure existing pipelines aren't broken.

Reply:
I didn't mean augmenting `sendtohelix.proj` -- I meant, calling into `sendtohelix.proj` from `sendtohelix-scenarios.proj`.

So basically `sendtohelix-scenarios` does the fan out of the scenarios and expands into `sendtohelix.proj` foreach scenario present. Then that way we avoid much duplication.

Then stuff specific to multiple scenarios like env vars and stuff like that, we could put it in a targets file and import that file when `scenario` is not null in `sendtohelix.proj`. Does that makes it easier/cleaner?

Reply:
I'll look into this; I've thought before about whether that would be easier, but was following the previous model (this same basic code exists in the coreclr `sendtohelix` equivalent).

Reply:
Yeah, but now that we share the same sources, I think it makes sense to avoid duplication and this brings a lot of duplication, like the targets to zip the testhost and to build the helix workitems, etc.